libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/MapboxNavigationTest.java

```diff
@@ -28,199 +28,251 @@
 
 public class MapboxNavigationTest extends BaseTest {
 
-  private MapboxNavigation navigation;
+  @Test
+  public void sanityTest() {
+    MapboxNavigation navigation = buildMapboxNavigation();
 
-  @Before
-  public void before() throws Exception {
-    navigation = new MapboxNavigation(mock(Context.class), ACCESS_TOKEN, mock(NavigationTelemetry.class),
-      mock(LocationEngine.class));
+    assertNotNull("should not be null", navigation);
   }
 
   @Test
-  public void sanityTest() {
-    assertNotNull("should not be null", navigation);
+  public void sanityTestWithOptions() {
     MapboxNavigationOptions options = MapboxNavigationOptions.builder().build();
-    MapboxNavigation navigationWithOptions = new MapboxNavigation(mock(Context.class),
-      ACCESS_TOKEN, options, mock(NavigationTelemetry.class), mock(LocationEngine.class));
+    MapboxNavigation navigationWithOptions = buildMapboxNavigationWithOptions(options);
+
     assertNotNull("should not be null", navigationWithOptions);
   }
 
   @Test
   public void voiceMilestone_onInitializationDoesGetAdded() throws Exception {
+    MapboxNavigation navigation = buildMapboxNavigation();
+
     assertTrue(navigation.getMilestones().get(0).getIdentifier() == VOICE_INSTRUCTION_MILESTONE_ID);
```

Comment:
Normally is good to structure the tests following the triple A 👀 https://speakerdeck.com/guardiola31337/elegant-unit-testing-mobilization-2016?slide=40 http://wiki.c2.com/?ArrangeActAssert

Here we're including the `Act` inside the `Assert`. It'd be great to have them separated and take advantage of giving great names to variables so it's easy to understand what's going on.

This applies to other tests included in `MapboxNavigationTest`.